pkg/handlers/query.go

```diff
@@ -44,6 +44,10 @@ func Query(env *gabi.Env) http.HandlerFunc {
 		}
 
 		now := time.Now()
+		if now.After(env.Expiration) {
+			http.Error(w, "Instance expired", http.StatusUnauthorized)
+			return
+		}
```

Comment:
this step will be moved a few lines down, after the user authorization: https://github.com/app-sre/qontract-reconcile/pull/2364#issuecomment-1112118300